### ISSUE TITLE
Pretty URLs doesn't play nicely with the share function on iOS

### ISSUE THREAD

**Issue by lucaberta** [stephenou/fruitionsite#3]
Created by @quicklywilliam in the mayneyao gist:
https://gist.github.com/mayneyao/b9fefc9625b76f70488e5d8c2a99315d#gistcomment-3295292

Not sure if this is part of the new custom slugs feature, but I noticed that this doesn't play nicely with the share function on iOS. For example, @stephenou's website redirects to https://fruitionsite.com/fruitionsite.com 🤔

**Comment by lucaberta** [comment]
Hi @quicklywilliam I have not been able to replicate the issue you are referring to.

I am using iPadOs 13.4.1 and the share sheet used by sharing, say, https://fruitionsite.com/roadmap provided the correct URL in the clipboard.

Are you able to do further tests and debugging to prove the issue?

**Comment by stephenou** [comment]
Also not able to repro. @quicklywilliam lemme know what you encounter.

**Comment by quicklywilliam** [comment]
You are right, copying to the clipboard does the right thing for me as well. Sharing via Messages does not.

Here are my exact steps to reproduce on an iPhone running iOS 13.3.1:
1. In Safari, navigate to https://fruitionsite.com/roadmap
2. Tap the share icon
3. Tap the Messages icon
4. Add a recipient and hit send
5. Open the messages app and navigate to the message you just sent
6. Tap the link.

**Comment by stephenou** [comment]
Thanks for the steps! I was able to repro.

However, I think this is an iOS bug. I got the same thing with Product Hunt:

1. In Safari, open https://www.producthunt.com/
2. Click on any product (URL should say `producthunt.com/posts/...`)
3. Share in iMessage with the same steps as yours

**Comment by quicklywilliam** [comment]
@stephenou I got a different result from Product Hunt – it shared the root home page link instead of the product URL – the wrong link but not an _invalid_ link. Definitely could be related at the root cause but something odd appears to be going on with how fruitionsite is rewriting URLs.

**Comment by lucaberta** [comment]
> @stephenou I got a different result from Product Hunt – it shared the root home page link instead of the product URL – the wrong link but not an _invalid_ link. Definitely could be related at the root cause but something odd appears to be going on with how fruitionsite is rewriting URLs.

I too did a few tests, and it definitely looks like the Apple Messages share sheet logic is somewhat broken.

I also had the same issue on Product Hunt, it shared the top-level URL and not the specific page I was looking at, though I am not sure whether what I was looking at way maybe a pop-up in the browser window, to be fair.

Still, if I would copy to the clipboard or create a markup snippet in Drafts.app, the URL copied from Fruition’s website worked properly.

In my view, this is an iOS/iPadOS bug.
